src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -312,6 +313,20 @@ private string FinishFixingName(string name)
             if (!string.IsNullOrEmpty(WordSeparator))
             {
                 words = name.Split(new[] { WordSeparator }, StringSplitOptions.RemoveEmptyEntries);
+
+                if (CapitalizationScheme == Capitalization.AllUpper || CapitalizationScheme == Capitalization.AllLower)
```

Comment:
Why just in these two cases? If I had it set to PascalCase and a word separator, don't we still want the same behavior?

Reply:
Done for all cases